guvnor-ala/guvnor-ala-services-rest/src/main/java/org/guvnor/ala/services/rest/RestPipelineServiceImpl.java

```diff
@@ -17,97 +17,156 @@
 import org.guvnor.ala.registry.PipelineRegistry;
-import org.guvnor.ala.registry.SourceRegistry;
+import org.guvnor.ala.registry.RuntimeRegistry;
+import org.guvnor.ala.runtime.providers.Provider;
+import org.guvnor.ala.runtime.providers.ProviderType;
 import org.guvnor.ala.services.api.PipelineService;
 import org.guvnor.ala.services.api.itemlist.PipelineConfigsList;
 import org.guvnor.ala.services.exceptions.BusinessException;
 
 @ApplicationScoped
 public class RestPipelineServiceImpl implements PipelineService {
 
-    @Inject
+    private PipelineExecutorTaskManager executorTaskManager;
+
     private PipelineRegistry pipelineRegistry;
 
-    @Inject
-    private SourceRegistry sourceRegistry;
+    private RuntimeRegistry runtimeRegistry;
 
-    @Inject
-    private BuildRegistry buildRegistry;
-
-    @Inject
-    @Any
-    private Instance<ConfigExecutor> configExecutors;
+    public RestPipelineServiceImpl() {
+        //Empty constructor for Weld proxying
+    }
 
     @Inject
-    private PipelineExecutor executor;
+    public RestPipelineServiceImpl(PipelineExecutorTaskManager executorTaskManager,
+                                   PipelineRegistry pipelineRegistry,
+                                   RuntimeRegistry runtimeRegistry) {
+        this.executorTaskManager = executorTaskManager;
+        this.pipelineRegistry = pipelineRegistry;
+        this.runtimeRegistry = runtimeRegistry;
+    }
 
-    @Inject
-    @Any
-    private Instance<PipelineEventListener> _eventListeners;
-
-    private PipelineEventListener[] eventListeners;
-
-    @PostConstruct
-    public void init() {
-        Iterator<ConfigExecutor> iterator = configExecutors.iterator();
-        Collection<ConfigExecutor> configs = new ArrayList<>();
-        while ( iterator.hasNext() ) {
-            ConfigExecutor configExecutor = iterator.next();
-            configs.add( configExecutor );
-        }
-        executor.init( configs );
-        final Collection<PipelineEventListener> eventListeners = new ArrayList<>();
-        for ( PipelineEventListener eventListener : _eventListeners ) {
-            eventListeners.add( eventListener );
-        }
-        this.eventListeners = eventListeners.toArray( new PipelineEventListener[]{} );
+    @Override
+    public PipelineConfigsList getPipelineConfigs(Integer page,
+                                                  Integer pageSize,
+                                                  String sort,
+                                                  boolean sortOrder) throws BusinessException {
+        final List<PipelineConfig> configs = new ArrayList<>();
+        pipelineRegistry.getPipelines(page,
+                                      pageSize,
+                                      sort,
+                                      sortOrder)
+                .stream()
+                .filter(p -> p.getConfig() != null)
+                .forEach((p) -> configs.add(p.getConfig()));
```

Comment:
ok, justy fyi, for debugging purposes you can use [Stream#peek(...)](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#i30)